pkg/providers/amifamily/bootstrap/args.go

```diff
@@ -0,0 +1,119 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/aws/karpenter-core/pkg/utils/resources"
+
+	"github.com/samber/lo"
+	core "k8s.io/api/core/v1"
+	"knative.dev/pkg/ptr"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+)
+
+// Options is the node bootstrapping parameters passed from Karpenter to the provisioning node
+type Options struct {
+	ClusterName             string
+	ClusterEndpoint         string
+	KubeletConfig           *v1alpha5.KubeletConfiguration
+	Taints                  []core.Taint      `hash:"set"`
+	Labels                  map[string]string `hash:"set"`
+	CABundle                *string
+	AWSENILimitedPodDensity bool
+	ContainerRuntime        *string
+	CustomUserData          *string
+}
+
+func (o Options) asKubeletExtraArgs() (args []string) {
+	args = append(args, o.nodeLabelArg(), o.nodeTaintArg(), o.maxPodsArg())
+
+	if o.KubeletConfig == nil {
+		return lo.Compact(args)
+	}
+	if o.KubeletConfig.PodsPerCore != nil {
+		args = append(args, fmt.Sprintf("--pods-per-core=%d", ptr.Int32Value(o.KubeletConfig.PodsPerCore)))
+	}
+	// We have to convert some of these maps so that their values return the correct string
+	args = append(args, joinParameterArgs("--system-reserved", resources.StringMap(o.KubeletConfig.SystemReserved), "="))
+	args = append(args, joinParameterArgs("--kube-reserved", resources.StringMap(o.KubeletConfig.KubeReserved), "="))
+	args = append(args, joinParameterArgs("--eviction-hard", o.KubeletConfig.EvictionHard, "<"))
+	args = append(args, joinParameterArgs("--eviction-soft", o.KubeletConfig.EvictionSoft, "<"))
+	args = append(args, joinParameterArgs("--eviction-soft-grace-period", lo.MapValues(o.KubeletConfig.EvictionSoftGracePeriod, func(v metav1.Duration, _ string) string { return v.Duration.String() }), "="))
+
+	if o.KubeletConfig.EvictionMaxPodGracePeriod != nil {
+		args = append(args, fmt.Sprintf("--eviction-max-pod-grace-period=%d", ptr.Int32Value(o.KubeletConfig.EvictionMaxPodGracePeriod)))
+	}
+	if o.KubeletConfig.ImageGCHighThresholdPercent != nil {
+		args = append(args, fmt.Sprintf("--image-gc-high-threshold=%d", ptr.Int32Value(o.KubeletConfig.ImageGCHighThresholdPercent)))
+	}
+	if o.KubeletConfig.ImageGCLowThresholdPercent != nil {
+		args = append(args, fmt.Sprintf("--image-gc-low-threshold=%d", ptr.Int32Value(o.KubeletConfig.ImageGCLowThresholdPercent)))
+	}
+	if o.KubeletConfig.CPUCFSQuota != nil {
+		args = append(args, fmt.Sprintf("--cpu-cfs-quota=%t", lo.FromPtr(o.KubeletConfig.CPUCFSQuota)))
+	}
+	return lo.Compact(args)
+}
+func (o Options) maxPodsArg() string {
+	if o.KubeletConfig != nil && o.KubeletConfig.MaxPods != nil {
+		return fmt.Sprintf("--max-pods=%d", ptr.Int32Value(o.KubeletConfig.MaxPods))
+	}
+	if !o.AWSENILimitedPodDensity {
+		return "--max-pods=110"
```

Comment:
So if the customer doesn't specify `max-pods`, it would be set as `110` right?
If so, this is in contrast with Windows where the default would be `500` going forward.